src/main/java/org/broadinstitute/hellbender/tools/walkers/genotyper/afcalc/ReferenceDiploidExactAFCalculator.java

```diff
@@ -0,0 +1,255 @@
+        }
+
+        // if we got here, then k > 0 for at least one k.
+        // the non-AA possible conformations were already dealt with by pushes from dependent sets;
+        // now deal with the AA case (which depends on previous cells in this column) and then update the L(j,k) value
+        for ( int j = 1; j < set.getLog10Likelihoods().length; j++ ) {
+            if ( totalK < 2*j-1 ) {
+                final double[] gl = genotypeLikelihoods.get(j);
+                final double conformationValue = MathUtils.log10(2*j-totalK) + MathUtils.log10(2*j-totalK-1) + set.getLog10Likelihoods()[j-1] + gl[HOM_REF_INDEX];
+                set.getLog10Likelihoods()[j] = MathUtils.approximateLog10SumLog10(set.getLog10Likelihoods()[j], conformationValue);
+            }
+
+            final double logDenominator = MathUtils.log10(2*j) + MathUtils.log10(2*j-1);
+            set.getLog10Likelihoods()[j] = set.getLog10Likelihoods()[j] - logDenominator;
+        }
+
+        double log10LofK = set.getLog10Likelihoods()[set.getLog10Likelihoods().length-1];
+
+        // update the MLE if necessary
+        stateTracker.updateMLEifNeeded(log10LofK, set.getACcounts().getCounts());
+
+        // apply the priors over each alternate allele
+        for ( final int ACcount : set.getACcounts().getCounts() ) {
+            if ( ACcount > 0 ) {
+                log10LofK += log10AlleleFrequencyPriors[ACcount];
+            }
+        }
+
+        stateTracker.updateMAPifNeeded(log10LofK, set.getACcounts().getCounts());
+    }
+
+    private static final class DependentSet {
+        public final int[] ACcounts;
+        public final int PLindex;
+
+        DependentSet(final int[] ACcounts, final int PLindex) {
+            this.ACcounts = ACcounts;
+            this.PLindex = PLindex;
+        }
+    }
+
+
+    // adds the ExactACset represented by the ACcounts to the ACqueue if not already there (creating it if needed) and
+    // also pushes its value to the given callingSetIndex.
+    private static void updateACset(final int[] newSetCounts,
+                                    final int numChr,
+                                    final ExactACset dependentSet,
+                                    final int PLsetIndex,
+                                    final Queue<ExactACset> ACqueue,
+                                    final Map<ExactACcounts, ExactACset> indexesToACset,
+                                    final List<double[]> genotypeLikelihoods) {
+        final ExactACcounts index = new ExactACcounts(newSetCounts);
+        if ( !indexesToACset.containsKey(index) ) {
+            final ExactACset set = new ExactACset(numChr/2 +1, index);
+            indexesToACset.put(index, set);
+            ACqueue.add(set);
+        }
+
+        // push data from the dependency to the new set
+        pushData(indexesToACset.get(index), dependentSet, PLsetIndex, genotypeLikelihoods);
+    }
+
+    private static void pushData(final ExactACset targetSet,
```

Comment:
note to self: port improvements from https://github.com/broadinstitute/gsa-unstable/pull/1270